File: 1016-subarray-sums-divisible-by-k/subarray-sums-divisible-by-k.py
class Solution(object):
    def subarraysDivByK(self, nums, k):
     d={0:1}
     prefix=0
     c=0
     for i in nums:
        prefix+=i
        if prefix%k in d:
            c+=d[prefix%k]
        d[prefix%k]=d.get(prefix%k,0)+1
     return c

File: 1016-subarray-sums-divisible-by-k/subarray-sums-divisible-by-k.java
class Solution {
    public int subarraysDivByK(int[] nums, int k) {
       HashMap<Integer,Integer> freq=new HashMap<>();
       freq.put(0,1);
       int sum=0;
       int c=0;
       for(int i=0;i<nums.length;i++){
        sum+=nums[i];
        int rem=((sum%k)+k)%k;
        if(freq.containsKey(rem)){
          c+=freq.get(rem);
        }
        freq.put(rem,freq.getOrDefault(rem,0)+1);
       }
    return c; 
    }
}